components/ui/form/Schedule.tsx

```diff
@@ -62,6 +65,34 @@ const TimeRangeField = ({ name }: TimeRangeFieldProps) => {
     );
   }, []);
 
+  const [valueSelectStart, setValueSelectStart] = useState<Option>(initialValueforSelectStart);
+
+  const createDateAndModifyNewValue = (onChange: any, option: any) => {
+    onChange(new Date(option?.value as number));
+    setValueSelectStart(option);
+  };
+
+  const calculateTimeRange = (value: Option) => {
+    const earlyMorningHours = [{ open: "00:00", close: "04:45" }];
+    let opened = false;
+    earlyMorningHours.forEach((item) => {
+      const open = new Date("1/1/1999 " + item.open);
+      const close = new Date("1/1/1999 " + item.close);
+      const hourDate = new Date(`1/1/1999 ${value.label}`);
+      opened = opened || (hourDate >= open && close >= hourDate);
```

Comment:
I think you can refactor this `forEach` to not run through all elements but to break this once the condition `(hourDate >= open && close >= hourDate)` is true.